src/React/Basic/Hooks/Aff.purs

```diff
@@ -1,31 +1,35 @@
 module React.Basic.Hooks.Aff where
 
 import Prelude
-
 import Data.Either (Either)
 import Data.Maybe (Maybe(..))
+import Data.Newtype (class Newtype)
 import Effect.Aff (Aff, error, killFiber, launchAff_, runAff)
 import Effect.Exception (Error)
-import React.Basic.Hooks (Hook, UseEffect, UseState, useEffect, useState, (/\))
+import React.Basic.Hooks (Hook, UseEffect, UseState, newtypeHook, useEffect, useState, (/\))
 import React.Basic.Hooks as React
 
-type UseAff key a hooks =
-  UseEffect key (UseState (Result a) hooks)
+newtype UseAff key a hooks
+  = UseAff (UseEffect key (UseState (Result a) hooks))
+
+derive instance ntUseAff :: Newtype (UseAff key a hooks) _
```

Comment:
This newtype plus the `newtypeHook` below results in `UseAff` appearing in type holes/errors rather than the inner type